tests/integration/local/start_api/test_start_api.py

```diff
@@ -1629,3 +1629,99 @@ def test_swagger_got_parsed_and_api_is_reachable_and_payload_version_is_2(self):
         self.assertEqual(response_data.get("version", {}), "2.0")
         self.assertIsNone(response_data.get("multiValueHeaders"))
         self.assertIsNotNone(response_data.get("cookies"))
+
+
+class TestWarmContainers(StartApiIntegBaseClass):
+    template_path = "/testdata/start_api/template-warm-containers.yaml"
+    container_mode = "WARM"
+
+    def setUp(self):
+        self.url = "http://127.0.0.1:{}".format(self.port)
+
+    @pytest.mark.flaky(reruns=3)
+    @pytest.mark.timeout(timeout=600, method="thread")
+    def test_all_containers_are_running(self):
```

Comment:
This test case validate that:
- all containers are initialized before calling any lambda function
- there is no new container will be created regardless which/how many lambda function will be invoked.

To split this test case, I need to start different local start-api command for each test case, if these test cases got executed in any order, they will give the same result. so, instead of doing that, I refactored this test case to make it more readable, but if you still think it needs to be divided, I will do it.

Reply:
So by biggest concern here is we hide failures. In a TDD framework, you want to know all the failures at once. This allows you to know the impact of your change. I generally prefer isolated tests as much as possible. I recognize this might not always be possible but something we should try to do everywhere we can.

Would it makes sense to have one Test Class for the first and another for the second? You are testing different use cases. Classes are cheap and can be used to organize different things. Maybe the problem is our naming of the containers? Maybe we shouldn't need to name them at all?